app/services/katello/product_content_importer.rb

```diff
@@ -35,12 +76,13 @@ def add_product_content(product, product_content_json)
     end
 
     def find_product_for_content(content_id)
-      prod = @cp_products.find do |prod_json|
-        prod_json['productContent'].any? do |product_content_json|
+      #Hash.find converts hash into ennuberable iterating over each key,value pair which we can capture based on find result
+      cp_product_id, __cp_product_value = @cp_products.find do |prod_json|
+        prod_json&.last&.first&.[]('productContent')&.any? do |product_content_json|
```

Comment:
Curious why the `&.last` needed to be added?

Reply:
The  `@cp_products.find` create an enumerable in the form an array which is [key, value] for the hash key:value pair and works on that. The last is to target the value in that. Weird formatting really on the return + the way .find iterates over it.